components/style/values/specified/align.rs

```diff
@@ -340,77 +340,78 @@ impl Parse for JustifyItems {
         if let Ok(value) = parse_overflow_self_position(input) {
             return Ok(JustifyItems(value))
         }
-        Err(())
+        Err(StyleParseError::UnspecifiedError.into())
     }
 }
 
 // auto | normal | stretch | <baseline-position>
-fn parse_auto_normal_stretch_baseline(input: &mut Parser) -> Result<AlignFlags, ()> {
+fn parse_auto_normal_stretch_baseline<'i, 't>(input: &mut Parser<'i, 't>)
+                                              -> Result<AlignFlags, ParseError<'i>> {
     if let Ok(baseline) = input.try(parse_baseline) {
         return Ok(baseline);
     }
 
     let ident = input.expect_ident()?;
-    match_ignore_ascii_case! { &ident,
+    (match_ignore_ascii_case! { &ident,
         "auto" => Ok(ALIGN_AUTO),
         "normal" => Ok(ALIGN_NORMAL),
         "stretch" => Ok(ALIGN_STRETCH),
         _ => Err(())
-    }
+    }).map_err(|()| BasicParseError::UnexpectedToken(Token::Ident(ident)).into())
```

Comment:
I'm assuming this funky way of doing this is because `ident` is borrowed in the `match` expression, is that right? If so, that's kind of unfortunate, I guess... :(